src/librustdoc/config.rs

```diff
@@ -548,16 +551,25 @@ impl Options {
             dcx.fatal("the `--test` flag must be passed to enable `--no-run`");
         }
 
+        let mut output_to_stdout = false;
         let test_builder_wrappers =
             matches.opt_strs("test-builder-wrapper").iter().map(PathBuf::from).collect();
         let out_dir = matches.opt_str("out-dir").map(|s| PathBuf::from(&s));
-        let output = matches.opt_str("output").map(|s| PathBuf::from(&s));
+        let output = matches.opt_str("output");
         let output = match (out_dir, output) {
-            (Some(_), Some(_)) => {
-                dcx.fatal("cannot use both 'out-dir' and 'output' at once");
+            (Some(_), Some(output)) => {
+                if output == "-" {
```

Comment:
What's the purpose of this logic? `rustdoc --out-dir something --output -` should be an error, I think.

Reply:
Because `--out-dir` is used for all rustdoc-ui tests, preventing this option to be tested. I thought about whether I should add a new flag to tests to allow removing this argument or to allow to have `--out-dir` and `--output -` used at the same time and considering it doesn't generate a file, I thought it would be ok. The only thing that I'm not clear about is whether or not it should a warning.